src/pages/settings/company/edit/CompanyEdit.tsx

```diff
@@ -0,0 +1,205 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Card, Element } from '@invoiceninja/cards';
+import { Button, InputField, SelectField } from '@invoiceninja/forms';
+import { AxiosError } from 'axios';
+import { endpoint } from 'common/helpers';
+import { request } from 'common/helpers/request';
+import { route } from 'common/helpers/route';
+import { toast } from 'common/helpers/toast/toast';
+import { useCurrentCompany } from 'common/hooks/useCurrentCompany';
+import { useLanguages } from 'common/hooks/useLanguages';
+import { CompanyInput } from 'common/interfaces/company-interface';
+import { ValidationBag } from 'common/interfaces/validation-bag';
+import { CurrencySelector } from 'components/CurrencySelector';
+import { LightSwitch } from 'components/LightSwitch';
+import { Modal } from 'components/Modal';
+import {
+  FormEvent,
+  useState,
+  SetStateAction,
+  Dispatch,
+  useEffect,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface Props {
+  isModalOpen: boolean;
+  setIsModalOpen: Dispatch<SetStateAction<boolean>>;
+}
+
+export function CompanyEdit(props: Props) {
+  const [t] = useTranslation();
+
+  const languages = useLanguages();
+
+  const currentCompany = useCurrentCompany();
+
+  const [errors, setErrors] = useState<ValidationBag>();
+
+  const [isFormBusy, setIsFormBusy] = useState<boolean>(false);
+
+  const [companyId, setCompanyId] = useState<string | undefined>();
+
+  const [company, setCompany] = useState<CompanyInput>();
+
+  const fetchCompanyDetails = async (company_id: string) => {
+    const response = await request(
+      'GET',
+      endpoint('/api/v1/companies/:id', { id: company_id })
+    );
+
+    const companySettings = response?.data?.data?.settings;
+
+    const { name, subdomain, language_id, currency_id } = companySettings;
+
+    setCompany({
+      name,
+      subdomain,
+      language_id,
+      currency_id,
+    });
+  };
+
+  const handleChange = (
+    property: keyof CompanyInput,
+    value: CompanyInput[keyof CompanyInput]
+  ) => {
+    setCompany((prevState) => ({
+      ...prevState,
+      [property]: value,
+    }));
+  };
+
+  const handleSave = async (event: FormEvent<HTMLFormElement>) => {
+    if (!isFormBusy) {
+      event?.preventDefault();
+
+      toast.processing();
+
+      setErrors(undefined);
+
+      setIsFormBusy(true);
+
+      const { name, language_id, subdomain, currency_id } = company || {};
```

Comment:
Why fallback to empty `{}`?  At this point, we need to make sure company is actual object, no undefined/empty.

Reply:
> Why fallback to empty `{}`? At this point, we need to make sure company is actual object, no undefined/empty.

Yeah sure, in this moment, object is 100% set. To resolve TS error, I just set {} as default value because without that it will indicate that company can be undefined. Fixed.